### PR TITLE
SAMZA-1602 : Moving class StreamAssert from src/test to src/main

### PR DESCRIPTION
**Pull request by Sanil15**
Tested with ./gradlew clean build, which is passing

### CODE REVIEW

File: samza-test/src/test/java/org/apache/samza/test/operator/BroadcastAssertApp.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 import org.apache.samza.operators.StreamGraph;
 import org.apache.samza.serializers.JsonSerdeV2;
 import org.apache.samza.test.operator.data.PageView;
-import org.apache.samza.test.util.StreamAssert;
+import com.apache.samza.test.framework.StreamAssert;
```
Comment (prateekm) [inline samza-test/src/test/java/org/apache/samza/test/operator/BroadcastAssertApp.java]: Why is this com.apache.samza?

Also, can you explain why this change is necessary? Is this expected to be used somewhere else?

Comment (Sanil15) [reply samza-test/src/test/java/org/apache/samza/test/operator/BroadcastAssertApp.java]: Updated com.apache.samza to org.apache.samza, my bad with the naming confusion.

Necessity for this change:

The StreamAsset class will be part of bigger framework that helps in Continuous integration testing that can be used by Open Source users as well as Samza-Li. Hence we move this out of test folder
